src/suggestion.rs

```diff
@@ -50,14 +69,206 @@ impl fmt::Display for Detector {
     }
 }
 
+// For long lines, literal will be trimmed to display in one terminal line.
+// Misspelled words that are too long shall also be ellipsized.
+pub fn condition_display_content(
+    terminal_size: usize,
+    _indent: usize,
+    stripped_line: &str,
+    mistake_range: Range,
+    padding_till_excerpt_start: usize,
+    marker_size: usize,
+) -> (String, usize, usize) {
+    if stripped_line.chars().count() + padding_till_excerpt_start <= terminal_size {
+        return (stripped_line.to_owned(), mistake_range.start, marker_size);
+    }
+
+    // The paddings give some space for the ` {} ...` and extra indentation and formatting:
+    //
+    //|---------------------------------------------------------------------------------------| terminal_size
+    //|-------| padding_till_excerpt_start = indent (3+line_number_digit_count) + 2 white spaces = 7usize, for this case.
+    //     |------| offset = PADDING_OFFSET; 3 chars for `...` and 2 white spaces more added in the formatting.
+    //
+    //   --> /home/tmhdev/Documents/cargo-spellcheck/src/suggestion.rs:62
+    //    |
+    // 62 |  ... Reasn of food, what's up with pie? There's strawberry pie, apple, pumpkin ...
+    //    |      ^^^^^
+    //    | - there, Cherie, thither, or tither
+    //    |
+    //    |   Possible spelling mistake found.
+    //
+    const MAX_MISTAKE_LEN: usize = 20;
+
+    const HEAD_DISPLAY_LEN: usize = 4;
+    const TAIL_DISPLAY_LEN: usize = 4;
+    const DOTS_LEN: usize = 3;
+
+    assert!(HEAD_DISPLAY_LEN + DOTS_LEN + TAIL_DISPLAY_LEN <= MAX_MISTAKE_LEN);
+
+    const TOTAL_CONTEXT_CHAR_COUNT: usize = 10;
+
+    // We will be using ranges to help doing the fitting:
+    //
+    // |-----------------------------------excerpt--------------------------------------|
+    // |----------------------|---------misspelled_word---------|-----------------------|
+    // |-----left_context-----|head_sub_range|-tail_sub_range-|-----right_context-----|
+    //
+    // Obs: paddings are not being considered in the illustration, but info is above.
+
+    // Misspelled words that are too long will be shrunken by ellipsizing parts of it.
+    let (marker_size, shrunken) = if mistake_range.len() > MAX_MISTAKE_LEN {
```

Comment:
shrunken is [scaring](https://www.google.com/search?q=shrunken&tbm=isch&ved=2ahUKEwjUi_vOyebqAhUI_4UKHWDdCI4Q2-cCegQIABAA&oq=shrunken&gs_lcp=CgNpbWcQA1AAWABgxhZoAHAAeACAAQCIAQCSAQCYAQCqAQtnd3Mtd2l6LWltZw&sclient=img&ei=CzAbX5TaK4j-lwTguqPwCA&bih=949&biw=1920
)

less scaring: 

- shorten
- lessen
- reduce
- decrease
- dwindle
- narrow
- drop off
- condense
- deflate
- shrivel
- wither
- make smaller
- truncate
- abbreviate
- compress
- squeeze

`+ mistake`

Just a suggestion!